src/App.js

```diff
@@ -29,14 +31,35 @@ console.log = function getMessages(message) {
   }
 };
 
-export default function App() {
-  const classes = useStyles();
+export async function decodeFSH(encodedFSH) {
+  if (encodedFSH.text === undefined) {
+    return 'Edit FSH here!';
+  } else {
+    const promisedURL = await expandLink(encodedFSH);
+
+    // Removes the encoded data from the end of the url, starting at index 38
+    const sliced64 = promisedURL.long_url.slice(40);
+    const displayText = inflateSync(Buffer.from(sliced64, 'base64')).toString('utf-8');
```

Comment:
I think I just hit a random edge case, but I had some nonsense typed into the id part of `.../share/:id` and the `promisedURL.long_url` I got back was not a FSH Online URL, so then I got an error when we try to slice and decode the text (error below). I think we might want to guard against this case, even if it is rare (you'd only get it if you were typing in a url and made a mistake probably). Maybe we can check that the `promisedURL.long_url` starts with `https://fshschool.org/FSHOnline/#/share/` or that `sliced64` isn't an empty string. Or maybe both. In those cases, we probably just want to return the default "Edit FSH here!" string. Either that, or maybe some of this manipulation should happen in the `expandLink` function in the `then`.

```
index.js:352 Uncaught (in promise) Error: unexpected end of file
    at Zlib._handle.onerror (index.js:352)
    at Zlib.push../node_modules/browserify-zlib/lib/binding.js.Zlib._error (binding.js:283)
    at Zlib.push../node_modules/browserify-zlib/lib/binding.js.Zlib._checkError (binding.js:238)
    at Zlib.push../node_modules/browserify-zlib/lib/binding.js.Zlib._write (binding.js:116)
    at Zlib.push../node_modules/browserify-zlib/lib/binding.js.Zlib.writeSync (binding.js:77)
    at Inflate.push../node_modules/browserify-zlib/lib/index.js.Zlib._processChunk (index.js:511)
    at zlibBufferSync (index.js:236)
    at push../node_modules/browserify-zlib/lib/index.js.exports.inflateSync (index.js:161)
    at decodeFSH (App.js:44)
    at async waitForFSH (App.js:60)
```